nncf/experimental/onnx/datasets/imagenet_dataset.py

```diff
@@ -35,25 +43,63 @@ def __len__(self):
         return len(self.dataset)
 
 
-def create_imagenet_torch_dataset(dataset_dir: str,
-                                  input_shape: List[int],
-                                  mean=(0.485, 0.456, 0.406),
-                                  std=(0.229, 0.224, 0.225),
-                                  crop_ratio=0.875,
-                                  batch_size: int = 1,
-                                  shuffle: bool = True):
-    import torchvision
-    from torchvision import transforms
-    image_size = [input_shape[-2], input_shape[-1]]
+def infer_input_shape(model: ModelProto) -> Tuple[Optional[int], int, int, int]:
+    assert len(model.graph.input) == 1
+
+    for _input in model.graph.input:
+        dim = _input.type.tensor_type.shape.dim
+        input_shape = [MessageToDict(d).get("dimValue") for d in dim]
+
+    input_shape = [int(dim) if dim is not None else dim for dim in input_shape]
+
+    assert len(input_shape) == 4
+
+    return input_shape
+
+
+def get_transform(image_size: Tuple[int, int],
```

Comment:
If we need to have `get_transform`, how about making this to common API for datasets instead of `imagenet_dataset.py`. Some APIs defined here could be shared with `detection_dataset.py` or `segmentation_dataset.py`.
What do you think?

Reply:
I'll be looking at it in the future PR which includes PTQ API benchmarks for ONNX Model ZOO object detection and segmentation models.